central/graphql/resolvers/node_components.go

```diff
@@ -0,0 +1,172 @@
+	utils.Must(
+		// NOTE: This list should be alphabetically ordered
+		schema.AddType("NodeComponent", []string{
+			"fixedIn: String!", // is this needed?
+			"id: ID!",
+			"lastScanned: Time",
+			"license: License",
+			"location(query: String): String!",
+			"name: String!",
+			"nodes(query: String, scopeQuery: String, pagination: Pagination): [Node!]!",
+			"nodeCount(query: String, scopeQuery: String): Int!",
+			"plottedVulnerabilities(query: String): PlottedVulnerabilities!",
+			"priority: Int!",
+			"riskScore: Float!",
+			"source: String!", // is this infrastructure ?
+			"topVulnerability: NodeVulnerability",
+			"unusedVarSink(query: String): Int",
+			"version: String!",
+			"vulnerabilities(query: String, scopeQuery: String, pagination: Pagination): [NodeVulnerability]!",
+			"vulnerabilityCount(query: String, scopeQuery: String): Int!",
+			"vulnerabilityCounter(query: String): VulnerabilityCounter!",
+		}),
+		// TODO : Tie this to NodeScan resolvers
+		// schema.AddExtraResolver("NodeScan", `nodeComponents(query: String, pagination: Pagination): [NodeComponent!]!`),
+		// schema.AddExtraResolver("NodeScan", `componentCount(query: String): Int!`),
+		schema.AddQuery("nodeComponent(id: ID): NodeComponent"),
+		schema.AddQuery("nodeComponents(query: String, scopeQuery: String, pagination: Pagination): [NodeComponent!]!"),
+		schema.AddQuery("nodeComponentCount(query: String): Int!"),
+	)
+}
+
+// NodeComponentResolver represents a generic resolver of node component fields.
+// Values may come from either an embedded component context, or a top level component context.
+// NOTE: This list should be alphabetically ordered
+type NodeComponentResolver interface {
+	FixedIn(ctx context.Context) string
+	ID(ctx context.Context) graphql.ID
+	LastScanned(ctx context.Context) (*graphql.Time, error)
+	License(ctx context.Context) (*licenseResolver, error)
+	Location(ctx context.Context, args RawQuery) (string, error)
+	Name(ctx context.Context) string
+	Nodes(ctx context.Context, args PaginatedQuery) ([]*nodeResolver, error)
+	NodeCount(ctx context.Context, args RawQuery) (int32, error)
+	PlottedVulnerabilities(ctx context.Context, args RawQuery) (*PlottedVulnerabilitiesResolver, error)
+	Priority(ctx context.Context) int32
+	RiskScore(ctx context.Context) float64
+	Source(ctx context.Context) string
+	TopVulnerability(ctx context.Context) (NodeVulnerabilityResolver, error)
+	UnusedVarSink(ctx context.Context, args RawQuery) *int32
+	Version(ctx context.Context) string
+	Vulnerabilities(ctx context.Context, args PaginatedQuery) ([]NodeVulnerabilityResolver, error)
+	VulnerabilityCount(ctx context.Context, args RawQuery) (int32, error)
+	VulnerabilityCounter(ctx context.Context, args RawQuery) (*VulnerabilityCounterResolver, error)
+}
+
+// NodeComponent returns a node component based on an input id (name:version)
+func (resolver *Resolver) NodeComponent(ctx context.Context, args IDQuery) (NodeComponentResolver, error) {
```

Comment:
Should have conditional branching on feature flag with an error message on Postgres case

Reply:
I will add those after testing whether regex conjunction queries work.